### DISCUSSION TITLE
Feature Request: Dashcam Mode

### DISCUSSION THREAD

**Discussion by thezeroth** [gopro/labs#304] [Feature Requests]
GoPro can be used as a dashcam, its good for GoPro business as it opens the door to a new market, people are more likely to buy a GoPro if they could also use it as a dashcam when they are not using it as an action camera.
In fact a GoPro is already an almost perfect dashcam, considering the capture quality and mounting options already available. the only thing that is missing is a little bit of code.

I imagine it this way: I show the GoPro a little barcode and it goes into Dashcam mode, it starts recording as soon as it receives power from the USB port and stops recording a certain amount after it stops receiving power from the USB Port; while recording it saves the video into smaller files while deleting the older files to empty space for new video.

We can already make the GoPro start and stop recording depending on power status of USB port, so the only part missing is to get the GoPro to save video in smaller files while deleting the older files to empty space.

I own a GoPro MAX and would love to see this feature available on my GoPro.

**Comment by dnewman-gpsw** [comment]
All of the above already exists on HERO11 + Mini Labs, including the optional deleting of older files to preserve capture space.  I'm not sure it will port to MAX, but there is a Labs update planned for that camera.

**Reply by thezeroth** [reply]
GoPro MAX is your latest product in its class and currently can not do looping.

1) Do you know why looping is not available on GoPro MAX and is it possible to make it available through the lab?

2) What are the chances that your company would add a "Dashcam Mode" (Maybe better to call it "Black Box Mode") to the main firmware (GoPro MAX currently has "Video", "Photo", "TimeWarp" modes), so user could place the GoPro in a specific place connected to USB power to capture an event that could happen at unknown time and not worry about running out of space on SD Card. Adding such a mode via firmware update would encourage your clients to retire their older GoPro to be their dashcam and then buy your newer models therefore boosting sales.

**Reply by dnewman-gpsw** [reply]
1. It is not possible to add new capture mode with Labs.
2. Unlikely.  

Labs is always about using existing features in enhanced ways, through experimental triggers or expanded exposure controls.  A native Dashcam mode would be a fundamental change to a camera like MAX.  Labs scripting is the closest you can get. You will have to wait and see if the scripting in the upcoming MAX labs is advanced enough to do this.  Generally the more advanced features are developed for the latest camera models, but MAX is getting some Labs attention.

**Reply by thezeroth** [reply]
following up on this feature request, since the last comment, we had 2 updates to GoPro MAX lab firmware, here is the change log:

> 2.02.70 - July 12, 2024
> Improved QR Code scanning on complex code in 360 mode.
> Improved the Labs fonts for LCD and Overlays.
> Latest logic for scripting
> Support for $ and * commands (etc. *BITR=90 and $FEAT=5)
> Added *WAKE=2 support (boot on power)
> 
> 2.00.75 - Apr 27, 2023
> Updates to the new Labs script language recently released for HERO11 Black and Mini. This allows for If-This-Then-That camera for customize camera triggers using a good range of system properties like: acceleration and gyroscope levels, battery level, GPS distance and precision, camera speed, loop count, scene motion and recording and USB status.
> Added support for BITR command for enhanced bitrate control. 360 HEVC can be increased from 60 up to 100Mb/s (90Mb/s good for most SD media)
> Added support for WAKE command, for auto boot the camera upon USB power.
> Added support for small logo overlays in 360 (and color bars for testing)
> improvements to motion detection
> improvements to histogram
> many smaller bug fixes for Labs

I am trying to figure out if using GoPro MAX as a dashcam is already a possibility, and if not what is missing.
1. does Clear SD up to XX GB command work for MAX?
2. can I run "Clear SD up to XX GB" while recording so I don't run out of space on the SD card?

**Reply by dnewman-gpsw** [reply]
1. Yes, MAX supports this feature.
2. It was not designed for clearing and recording at the same time.  It might work, but it will very likely stop the current recording.  You can try it.
